qdev_wrappers/sweep_functions.py

```diff
@@ -30,16 +30,11 @@ def _flush_buffers(*params):
     Supposed to be called inside doNd like so:
     _flush_buffers(inst_set, *inst_meas)
     """
-    instrument_names = set()
-    for param in params:
-        if hasattr(param, '_instrument'):
-            instrument_names.add(param._instrument.root_instrument.name)
-        elif isinstance(param, VisaInstrument):
-            instrument_names.add(param.root_instrument.name)
-
-    for name in instrument_names:
+    for name in set(p.root_instrument.name
+                    for p in params
+                    if p.root_instrument is not None):
         instr = qc.Instrument.find_instrument(name)
-        if isinstance(instr, qc.VisaInstrument):
+        with suppress(AttributeError):
```

Comment:
Can we add a comment to why we need to suppress this Error for future reference

Reply:
That would certainly be good :-)